src/words/ast/INodeMovesPredicate.java

```diff
@@ -10,7 +15,22 @@ public INodeMovesPredicate(Object... children) {
 
 	@Override
 	public ASTValue eval(WordsEnvironment environment) throws WordsRuntimeException {
-		// TODO
-		throw new AssertionError("Not yet implemented");
+		ASTValue className = children.get(0).eval(environment);
+		ASTValue objectAlias = children.get(1).eval(environment);
+		ASTValue moveDirection = children.get(2) == null ? null : children.get(2).eval(environment);
+		
+		HashSet<WordsObject> objectsOfClass = environment.getObjectsByClass(className.stringValue);
+
+		for (WordsObject object : objectsOfClass) {
```

Comment:
Yeah, I think that's a good way to do it.  For boolean predicates, though, I think the listener itself might have to run its statement list, since boolean predicates appear elsewhere besides listeners.  But for basic action predicates, what you're saying makes sense.  Conceptually, a basic action predicate isn't exactly a predicate but a "template" for a predicate, so it makes sense that a listener would provide the template the code that should be executed for any instantiation of the template that is true.